### PR TITLE
test: force init argilla client credentials

### LINKED ISSUE #3228: [BUG] Unit test at `tests/datasets/test_datasets.py::test_settings_workflow` failing when running `pytest`
**Describe the bug**

As of the recent addition of the new `owner` role, it seems that a unit test is failing due to unknown reasons, most likely due to a residual test that's changing something unexpected, that's why we've decided it to skip `tests/datasets/test_datasets.py::test_settings_workflow` for the moment while we look for a solution.

**Stacktrace and Code to create the bug**

Simply running the whole test suite as `pytest` will eventually lead into this issue, but running `pytest tests/datasets/test_datasets.py::test_settings_workflow` won't fail.

<details>
<summary>How do I dropdown?</summary>
<pre>
FAILED tests/datasets/test_datasets.py::test_settings_workflow[settings_0-wrong_settings0] - argilla.client.sdk.commons.errors.UnauthorizedApiError: Argilla server returned an error with http status: 401
Error details: [{'code': 'argilla.api.errors::UnauthorizedError', 'params': {'detail': 'Could not validate credentials'}}]
</pre>
</details>

**Expected behavior**

The test should pass as it pass when ran as standalone.

**Environment:**
 - Argilla Version [e.g. 1.0.0]: 1.11.0.dev0
 - ElasticSearch Version [e.g. 7.10.2]: 7.17, 8, 8.5, 8.8
 - Docker Image (optional) [e.g. argilla:v1.0.0]: argilla:v1.11.0

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR fix the weird failing test by forcing an Argilla init with default user credentials before the test operations.

Closes #3228

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] Test fix

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

The test works now.

### CODE REVIEW

File: tests/datasets/test_datasets.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 from argilla.client.sdk.commons.errors import ForbiddenApiError
 from argilla.datasets import TextClassificationSettings, TokenClassificationSettings
 from argilla.server.contexts import accounts
-from argilla.server.security.model import WorkspaceUserCreate
+from argilla.server.security.model import User, WorkspaceUserCreate
```
Comment (alvarobartt) [inline tests/datasets/test_datasets.py:21]: This is not the correct import for the type-checking on `argilla_user`, it is `from argilla.server.models import User`

Comment (alvarobartt) [inline tests/datasets/test_datasets.py]: This is not the correct import for the type-checking on `argilla_user`, it is `from argilla.server.models import User`

Comment (alvarobartt) [reply tests/datasets/test_datasets.py]:
```suggestion
from argilla.server.models import User
from argilla.server.security.model import WorkspaceUserCreate
```

Check the `pre-commit` in advance, as not sure if that's the correct order (I think yes, but double check please)

**Review by alvarobartt — CHANGES_REQUESTED** [review]

File: tests/datasets/test_datasets.py (hunk 1 above)
Comment (frascuchon) [reply tests/datasets/test_datasets.py]: thanks. Done!

**Review by gabrielmbmb — APPROVED** [review]

**Review by alvarobartt — APPROVED** [review]

**Pull request merged**